rpcs3/rpcs3qt/main_window.cpp

```diff
@@ -525,12 +563,56 @@ void main_window::show_boot_error(game_boot_result status)
 	msg->setTextFormat(Qt::RichText);
 	msg->setStandardButtons(QMessageBox::Ok);
 	msg->setText(tr("Booting failed: %1 %2").arg(message).arg(link));
-	msg->setAttribute(Qt::WA_DeleteOnClose);
+	msg->setParent(this);
 	msg->open();
+
+	if (status != game_boot_result::still_running)
+	{
+		msg->setAttribute(Qt::WA_DeleteOnClose);
+		return;
+	}
+
+	// Automatically close for still_running error
```

Comment:
What does this complicated spaghetti code achieve and why is it necessary?

Reply:
Removed